bunsen/bunsen-core/src/main/java/com/cerner/bunsen/definitions/StructureDefinitions.java

```diff
@@ -53,17 +66,366 @@ public abstract class StructureDefinitions {
+
+    // Extensions may contain either additional extensions or a value field, but not both.
+    // So if it has a child element with a slice name, it has additional extensions.
+    List<IElementDefinition> childExtensions = children.stream()
+        .filter(element -> element.getSliceName() != null)
+        .collect(Collectors.toList());
+
+    if (!childExtensions.isEmpty()) {
+      List<StructureField<T>> childFields = new ArrayList<>();
+
+      for (IElementDefinition childExtension: childExtensions) {
+        List<StructureField<T>> childField = extensionElementToFields(visitor,
+            rootDefinition, childExtension, extensionDefinitions, stack);
+        childFields.addAll(childField);
+      }
+
+      T result = visitor.visitParentExtension(sliceName,
+          url,
+          childFields);
+
+      if (result == null) {
+        return Collections.emptyList();
+      } else {
+        return Collections.singletonList(
+            StructureField.extension(sliceName,
+                url,
+                extensionRoot.getIsModifier(),
+                result));
+      }
+    } else {
+      // The extension has no children, so produce its value.
+      Optional<IElementDefinition> valueElement = children.stream()
+          .filter(e -> e.getPath().contains("value"))
+          .findFirst();
+      // FIXME: get the extension URL.
+      Optional<IElementDefinition> urlElement = children.stream()
+          .filter(e -> e.getPath().endsWith("url"))
+          .findFirst();
+      String extensionUrl = urlElement.get().getFixedPrimitiveValue();
+      List<StructureField<T>> childField = elementToFields(visitor, rootDefinition,
+          valueElement.get(), extensionDefinitions, stack);
+      T result = visitor.visitLeafExtension(sliceName,
+          extensionUrl,
+          childField.iterator().next().result());
+      return Collections.singletonList(
+          StructureField.extension(sliceName,
+              extensionUrl,
+              extensionRoot.getIsModifier(),
+              result));
+    }
+  }
+
+  /**
+   * Returns the fields for the given element. The returned stream can be empty
+   * (e.g., for elements with max of zero), or have multiple values (for elements
+   * that generate fields with additional data in siblings.)
+   */
+  private <T> List<StructureField<T>> elementToFields(DefinitionVisitor<T> visitor,
+      IStructureDefinition rootDefinition,
+      IElementDefinition element,
+      List<IElementDefinition> snapshotDefinitions,
+      Deque<QualifiedPath> stack) {
+
+    String elementName = DefinitionVisitorsUtil.elementName(element.getPath());
+
+    if (element.getMax().equals("0")) {
+      // Fields with max of zero are omitted.
+      return Collections.emptyList();
+    } else if ("Extension".equals(element.getFirstTypeCode())) {
+      return extensionElementToFields(visitor, rootDefinition, element, snapshotDefinitions, stack);
+    } else if (element.getSliceName() != null) {
```

Comment:
Question : This check was earlier present only for `R4StructureDefinitions` and now we are generalising for all. I hope this is fine.

Reply:
Thanks for flagging this. I took a look and it seems this was added in PR #692 to fix #495 and as [this comment](https://github.com/google/fhir-data-pipes/issues/495#issuecomment-1548804592) explains, it was an issues for both R4 and STU3. I think not adding it to STU3 was a shortcoming of PR #692 (we should really add STU3 unit-tests for that particular case).